### PR TITLE
Add loading modal

### PR DESCRIPTION
**Pull request by pochretien**
### Potential Breaking Changes
I changed the composite modal Interface, title is now optional
No breaking change but juste make sure we use a title as prop for each composite modal to render a header. The title was required before

New component ModalLoading

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by pochretien** [comment]
Preview: 
image

File: packages/react-vapor/src/components/modal/loading/examples/ModalLoadingExamples.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    render() {
+        return (
+            // start-print
+            <div className="mt3">
+                <label className="form-control-label">Loading modal</label>
+                <div className="form-group">
+                    <button className="btn" onClick={() => this.openModal(Ids.example1)}>
+                        Open Modal
+                    </button>
+                    <ModalLoading id={Ids.example1} title={'my loading title'} />
```
Comment (gdostie) [inline packages/react-vapor/src/components/modal/loading/examples/ModalLoadingExamples.tsx]:
```suggestion
                    <ModalLoading id={Ids.example1} title={'my loading title'} openOnMount={false} />
```
When clicking on the `ModalLoading` section in the demo its annoying to have it opened in our face haha

File: packages/react-vapor/src/components/modal/loading/ModalLoading.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import {Loading} from '../../loading/Loading';
+import {ModalCompositeConnected} from '../ModalCompositeConnected';
+
+export interface ModalLoadingProps {
+    id: string;
+    title?: React.ReactNode;
+    openOnMount?: boolean;
+}
+
+export const ModalLoading = (props: ModalLoadingProps) => (
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/modal/loading/ModalLoading.tsx]:
```suggestion
export const ModalLoading: React.FunctionComponent<ModalLoadingProps> = (props) => (
```

File: packages/react-vapor/src/components/modal/tests/ModalComposite.spec.tsx [hunk 3]
```diff
@@ -46,16 +46,18 @@ describe('ModalComposite', () => {
...
 
     it('should render modal header children inside the modal header', () => {
         const modalHeaderChildren = <span>A bird in the hand is worth two in the bush.</span>;
-        const modalComposite = shallow(<ModalComposite isOpened modalHeaderChildren={modalHeaderChildren} />).dive();
+        const modalComposite = shallow(
+            <ModalComposite isOpened modalHeaderChildren={modalHeaderChildren} title={'patate'} />
+        ).dive();
 
         expect(modalComposite.find(ModalHeader).contains(modalHeaderChildren)).toBe(true);
     });
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/modal/tests/ModalComposite.spec.tsx]: You could add a test that validate there's no header when the title is null

File: packages/react-vapor/src/components/modal/loading/examples/ModalLoadingExamples.tsx (hunk 1 above)
Comment (pochretien) [reply packages/react-vapor/src/components/modal/loading/examples/ModalLoadingExamples.tsx]: Good point!
